File: Data Structures & Algorithms/two-integer-sum/submission-0.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_map = {}
        
        for index, value in enumerate(nums):
            complement = target - value
            if complement in index_map:
                return [index_map[complement], index]
            index_map[value] = index

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int [] res = new int [nums.length];

        for(int i=0;i<nums.length;i++)
        {
            int mul=1;
            for(int j=0;j<nums.length;j++)
            {
                if(i!=j)
                {
                    mul*=nums[j];


                }

            }
            res[i]=mul;
        }

        return res;
        
    }
}  
